### DISCUSSION TITLE
[@types/react] useCallback types of arguments with default values inferred incorrectly

### DISCUSSION THREAD

**Discussion by FridGrip** [DefinitelyTyped/DefinitelyTyped#57734] [Issues with a @types package]
Not sure why this happens, but if I have the following statements:

```
const a = (x = false) => {};
const b = useCallback((x = false) => {}, []);
```

`a` will have the correct type (`x?: boolean) => void;`) while `b` will lose its argument typing: `(x?: any) => void;`

[a comment by an automated account is not shown]

**Comment by wesley-at-hl** [comment]
I'm having this issue as well, though it seems like a typescript issue. A simplified version of this still have this issue
```ts
function useCallback<T, U>(callback: (val: T) => U): (val: T) => U {
    return callback
};
useCallback((val = true) => val) // val is unknown 
```
[see in ts playground](https://www.typescriptlang.org/play?ts=4.6.2#code/LAKAZgrgdgxgLgSwPZQAQQM4FMDCBDAGwICM8YBrAHgBUAaVAVQD4AKGQks8gLlRYDdCvagEpUAXiaMRvAUNSiJUhqgDeoVJtQAnLHAja07IqQqgAvgG5QoTLg6nyLOQQmo42iFhmpiSJARYeGiSqIIEIkA)
